builtin/credential/ldap/backend.go

```diff
@@ -132,6 +132,14 @@ func (b *backend) Login(ctx context.Context, req *logical.Request, username stri
 		return nil, logical.ErrorResponse(err.Error()), nil, nil
 	}
 
+	if cfg.AnonymousGroupSearch {
+		c.Close()
+		c, err = ldapClient.DialLDAP(cfg.ConfigEntry)
+		if err != nil {
+			return nil, logical.ErrorResponse("ldap operation failed"), nil, nil
```

Comment:
should you preserve the original "err" in the error response somehow?

Reply:
The response could be preserved but this is not a log error, rather it is (what will become) the http response, which in my opinion should not contain the whole trace but the concise error.